src/var.c

```diff
@@ -41,20 +41,18 @@
 #include <rdma/fi_var.h>
 #include <rdma/fi_log.h>
 
-struct fi_var_t {
-	const struct fi_provider *provider;
-	char *var_name;
-	char *help_string;
-
-	char *env_var_name;
-
-	struct fi_var_t *next;
-};
+#include "fi.h"
 
 static struct fi_var_t *fi_vars = NULL;
 
+__attribute__((visibility ("default")))
+struct fi_var_t *DEFAULT_SYMVER_PRE(_fi_getvars)(void) {
```

Comment:
> Yes probably. This is a public facing API but it should only be used by
> tooling or otherwise "special" users.

We need to design this with the expectation that some random app is going to try to use it for some obscure reason.  E.g. looking for some specific option that they can enable.

We definitely don't want to provide handles to internal data structures.  Maybe we need something more like:

struct fi_vars {
	const char *prov_name;
	const char *var_name;
	...
}

int fi_getvars(struct fi_vars *vars, int *count);
fi_freevars(vars);

We could drop fi_freevars() if we want the caller to allocate the memory, with the necessary size returned by fi_getvars().